Toolkit/ArcGISToolkit/TimeSlider.swift

```diff
@@ -0,0 +1,2609 @@
+        
+        // Notify that the value is changed
+        if !isContinuous {
+            notifyChangeOfCurrentExtent()
+        }
+    }
+    
+    // Refresh the slider when requried
+    public override func layoutSubviews() {
+        refresh()
+    }
+    
+    // Deinit
+    deinit {
+        removeObservers()
+    }
+    
+    // MARK: Public Functions
+    
+    /**
+     This will look for all time aware layers which are visible and are participating in time based filtering
+     to initialize slider's fullExtent, currentExtent and timeStepInterval properties.
+     */
+    public func initializeTimeProperties(geoView: AGSGeoView, completion: @escaping (Error?)->Void) {
+        //
+        // Set geoview and it's properties on the slider
+        // it will initialize required properties.
+        self.geoView = geoView
+
+        // Set operational layers
+        guard let operationalLayers = geoView.operationalLayers, !operationalLayers.isEmpty else {
+            completion(NSError(domain: AGSErrorDomain, code: AGSErrorCode.commonNoData.rawValue, userInfo: [NSLocalizedDescriptionKey : "There are no time aware layers to initialize time slider."]))
+            return
+        }
+        
+        // Set map/scene
+        if let mapView = geoView as? AGSMapView {
+            map = mapView.map
+        }
+        else if let sceneView = geoView as? AGSSceneView {
+            scene = sceneView.scene
+        }
+        
+        // Start observing
+        addObservers()
+        
+        //
+        // Loop through all time aware layers which are visible and are participating in time based filtering
+        // and initialize slider's fullExtent, currentExtent and timeStepInterval properties
+        var timeAwareLayersFullExtent: AGSTimeExtent?
+        var timeAwareLayersStepInterval: AGSTimeValue?
+        var supportsRangeTimeFiltering = true
+        
+        //
+        // This will help in waiting for all layer/sublayers
+        // to be loaded to gather requried information
+        // to load time properties
+        let dispatchGroup = DispatchGroup()
+        
+        // Load all operational layers to initialize slider properties
+        AGSLoadObjects(operationalLayers) { [weak self] (loaded) in
+            //
+            // Bail out if layers
+            // are not loaded
+            guard loaded else {
+                return
+            }
+            
+            // Make sure self is around
+            guard let strongSelf = self else {
+                return
+            }
+            
+            // Once, all layers are loaded,
+            // loop through all of them.
+            operationalLayers.forEach({ (layer) in
+                //
+                // The layer must be time aware, supports time filtering and time filtering is enabled.
+                if let timeAwareLayer = layer as? AGSTimeAware, timeAwareLayer.supportsTimeFiltering, timeAwareLayer.isTimeFilteringEnabled {
```

Comment:
@njarecha You might consider replacing this `if` statement with a `guard` statement.

Reply:
fixed